db/seeds/schedules_and_milestones.rb

```diff
@@ -0,0 +1,817 @@
+def describe_schedule_and_milestones(schedule:, year:, milestones:)
+  milestone_description = milestones
+    .map { |m| m[:declaration_type] }
+    .map do |dt|
+      case dt
+      when 'started'
+        Colourize.text(shorten(dt), :green)
+      when /retained/
+        Colourize.text(shorten(dt), :yellow)
+      when 'completed'
+        Colourize.text(shorten(dt), :red)
+      when /extended/
+        Colourize.text(shorten(dt), :cyan)
+      end
+    end
+
+  print_seed_info("Added schedule #{colourize_year(year)} - #{schedule.identifier} (#{milestone_description.join(', ')})", indent: 2)
+end
+
+def colourize_year(year)
+  (year.odd?) ? Colourize.text(year, :magenta) : year
+end
+
+def shorten(declaration_type)
+  first_char = declaration_type[0]
+  last_char = declaration_type[-1]
+
+  "#{first_char}#{last_char if last_char.match?(/\A\d+\Z/)}"
+end
+
+def schedule_and_milestone_data
+  [
+    {
+      identifier: "ecf-extended-september",
```

Comment:
there's a bit of duplication with the schedules especially extended/reduced/replacement. Maybe we can set one date for each of those types and reuse to avoid having large list of schedules. I understand though might be easier to read and change with duplication, happy to chat

Reply:
Yeah that sounds like a good idea. I built this directly from an export of ECF1 data converted to a big hash. I'll take a look tomorrow.